linera-views/src/backends/value_splitting.rs

```diff
@@ -255,52 +269,55 @@ where
                 }
             }
         }
-        self.store.write_batch(batch_new).await
+        Ok(self.store.write_batch(batch_new).await?)
     }
 
     async fn clear_journal(&self) -> Result<(), Self::Error> {
-        self.store.clear_journal().await
+        Ok(self.store.clear_journal().await?)
     }
 }
 
 impl<K> ValueSplittingStore<K>
 where
     K: RestrictedKeyValueStore + Send + Sync,
-    K::Error: From<bcs::Error> + From<DatabaseConsistencyError>,
+    K::Error: 'static,
 {
     /// Creates a new store that deals with big values from one that does not.
     pub fn new(store: K) -> Self {
         ValueSplittingStore { store }
     }
 
-    fn read_count_from_value(value: &[u8]) -> Result<u32, K::Error> {
+    fn read_count_from_value(value: &[u8]) -> Result<u32, <Self as WithError>::Error> {
         if value.len() < 4 {
-            return Err(DatabaseConsistencyError::NoCountAvailable.into());
+            return Err(ValueSplittingError::NoCountAvailable);
         }
         let mut bytes = value[0..4].to_vec();
         bytes.reverse();
         Ok(bcs::from_bytes::<u32>(&bytes)?)
     }
 
-    fn get_segment_key(key: &[u8], index: u32) -> Result<Vec<u8>, K::Error> {
+    fn get_segment_key(key: &[u8], index: u32) -> Result<Vec<u8>, <Self as WithError>::Error> {
         let mut big_key_segment = key.to_vec();
         let mut bytes = bcs::to_bytes(&index)?;
         bytes.reverse();
         big_key_segment.extend(bytes);
         Ok(big_key_segment)
     }
 
-    fn read_index_from_key(key: &[u8]) -> Result<u32, K::Error> {
+    fn read_index_from_key(key: &[u8]) -> Result<u32, <Self as WithError>::Error> {
         let len = key.len();
         if len < 4 {
-            return Err(DatabaseConsistencyError::TooShortKey.into());
+            return Err(ValueSplittingError::<K::Error>::TooShortKey);
```

Comment:
`ValueSplittingError::TooShortKey` should be enough

Reply:
Correct.